src/core/src/engine/consensus_state_store.rs

```diff
@@ -75,6 +76,49 @@ impl<'a> ConsensusStateStore<'a> {
         debug!("Stored state for block_id : {}", poet2_util::to_hex_string(&block_id));
         Ok(())
     }
+
+    pub fn add_to_state_store( &mut self, block_id: &BlockId, block: &Block,
```

Comment:
Question: Does this block_id correspond to block.block_id? If so can we reduce a parameter?

Reply:
That can be done for sure.